tests/30rooms/11leaving.pl

```diff
@@ -0,0 +1,299 @@
+multi_test "Setup a room, and have the first user leave (SPEC-216)",
+
+    requires => [qw(
+        make_test_room change_room_powerlevels do_request_json_for user
+        more_users can_create_room
+    )],
+
+    provides => [qw( departed_room_id )],
+
+    # User A creates a room.
+    # User A invites User B to the room.
+    # User B joins the room.
+    # User B will set the ("m.room.name", "") state of the room to {
+    #   "body": "N1. B's room name before A left"
+    # }
+    # User B will set the ("madeup.test.state", "") state of the room to {
+    #   "body": "S1. B's state before A left"
+    # }
+    # User B will send a message with body "M1. B's message before A left"
+    # User B will send a message with body "M2. B's message before A left"
+    # User A will leave the room.
+    # User B will set the ("m.room.name", "") state of the room to {
+    #   "body": "N2. B's room name after A left"
+    # }
+    # User B will set the ("madeup.test.state", "") state of the room to {
+    #   "body": "S2. B's state after A left"
+    # }
+    # User B will send a message with text "M3. B's message after A left"
+    #
+    do => sub {
+        my (
+            $make_test_room, $change_room_powerlevels, $do_request_json_for,
+            $user_a, $more_users
+        ) = @_;
+        my $user_b = $more_users->[1];
+
+        my $room_id;
+
+        $make_test_room->([$user_a, $user_b])->then( sub {
+            ( $room_id ) = @_;
+
+            provide departed_room_id => $room_id;
+
+            $change_room_powerlevels->($user_a, $room_id, sub {
+                my ( $levels ) = @_;
+                # Set user B's power level so that they can set the room
+                # name. By default the level to set a room name is 50.
+                $levels->{users}{ $user_b->user_id } = 50;
+            })
+        })->then( sub {
+            $do_request_json_for->( $user_b,
+                method => "PUT",
+                uri => "/api/v1/rooms/$room_id/state/m.room.name",
+                content => { "name" => "N1. B's room name before A left", },
+            )->on_done(sub { pass "User B set the room name the first time" })
```

Comment:
Since we're really not expecting that all of these preparation steps will fail (because we're only doing things that earlier test files have asserted work), there's no real benefit in having all the `pass` messages here. They might as well be removed to reduce the verbosity of output